src/components/map/LocationOverview.tsx

```diff
@@ -44,8 +46,24 @@ const mapToMapMarkers = (pointsOfInterest: any): MapMarker[] | undefined => {
 
         if (!latitude || !longitude) return undefined;
 
+        let icon = location;
+
+        if (item.category?.iconName) {
+          // remove location and locationIconAnchor from Icons for proper type perspective to all
+          // the other icons
+          // eslint-disable-next-line @typescript-eslint/no-unused-vars
+          const { location: _l, locationIconAnchor: _lia, ...OtherIcons } = Icons;
+          const iconName = item.category.iconName as keyof typeof OtherIcons;
+
+          // only use the icon if it is available from our icons
+          if (Object.keys(OtherIcons).includes(iconName)) {
+            icon = OtherIcons[iconName] as (color: any) => string;
+          }
+        }
+
         return {
-          icon: location(colors.primary),
+          icon: icon(colors.primary),
+          activeIcon: icon(colors.accent),
```

Comment:
i was able to reduce duplication here with using one previously set `icon`, that gets called with the color